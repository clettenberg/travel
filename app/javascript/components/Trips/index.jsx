import React from 'react'
import PropTypes from 'prop-types'

import Map from '../Map'
import styles from './style.module.scss'

const propTypes = {
  trips: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    places: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      point: PropTypes.arrayOf(PropTypes.number)
    }))
  })).isRequired
}

const Trips = (props) => {
  const points = props.trips.map(trip => (
    trip.places.map(place => place.point)
  )).reduce((a, b) => a.concat(b), [])
    .filter(Boolean)

  return (
    <div className={styles.container} >
      <Map
        className={styles.map}
        points={points}
        center={[-98.2518834, 39.8112442]}
        zoom={1}
      />
      <div className={`${styles.trips}`}>
        <ul className={`list-group list-group-flush`}>
          {props.trips.map(trip => (
            <a
              href={`/trips/${trip.id}`}
              key={trip.id}
              className='list-group-item list-group-item-action'
            >
              {trip.title}
            </a>
          ))}
        </ul>
        <a className={`${styles.sticky} btn btn-primary`} href='/trips/new'>Add Trip</a>
      </div>
    </div>
  )
}

Trips.propTypes = propTypes
export default Trips
