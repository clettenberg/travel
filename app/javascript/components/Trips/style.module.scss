$mobile-breakpoint: 768px;

.container {
  display: grid;

  @media (max-width: $mobile-breakpoint) {
    grid-template-columns: auto;
    grid-template-rows: [row1-start] 40vh [row1-end row2-start] 60vh [row2-end];
    grid-template-areas: "maps"
    "trips";
  }

  @media (min-width: $mobile-breakpoint) {
    grid-template-columns: 30vw auto;
    grid-template-rows: [row1-start] 100vh [row1-end];
    grid-template-areas: "trips maps";
  }
}

.trips {
  grid-area: trips;
  overflow: scroll;
}

.maps {
  grid-area: maps;
}

.sticky {
  position: fixed;
  position: sticky;
  bottom: 0px;
  z-index: 2;
  width: 100%;
}
