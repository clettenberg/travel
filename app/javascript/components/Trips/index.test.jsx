
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Trips from './index'

describe('Trips', () => {
  it('should render', () => {
    const wrapper = shallow(
      <Trips trips={
        [{
          id: 26,
          title: 'UK with Ron and Stacey',
          places: [{
            id: 286,
            name: 'The Pink House (Behind the caravan park)',
            point: [-5.21714480341615, 51.8780559],
            note: '',
            startDate: '09/20/19',
            endDate: '09/22/19'
          }, {
            id: 287,
            name: 'AirBnB in Edinburgh',
            point: [-3.1936512, 55.9499331],
            note: 'u003cdivu003eGreat flat in old Edinburghu003c/divu003e',
            startDate: '09/25/19',
            endDate: '09/28/19'
          }],
          center: [-2.6707358336225657, 53.941955750000005],
          startDate: '09/15/19',
          endDate: '09/28/19'
        },
        {
          id: 25,
          title: 'Europe 2019',
          places: [{
            id: 281,
            name: 'Lunch at Buenasmigas',
            point: [2.1773041, 41.3838014],
            note: 'u003cdivu003eLunch near the la sagrada familiau003c/divu003e',
            startDate: '09/01/19',
            endDate: '09/01/19'
          }, {
            id: 282,
            name: 'Sagrada FamÃ­lia ',
            point: [2.17439826359357, 41.4034719],
            note: 'u003cdivu003eTour of the cathedral. Went up in the tower.u0026nbsp;u003c/divu003e',
            startDate: '09/01/19',
            endDate: '09/01/19'
          }],
          center: [11.00602877486278, 41.65760095],
          startDate: '09/01/19',
          endDate: '09/15/19'
        }]
      }
      />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
