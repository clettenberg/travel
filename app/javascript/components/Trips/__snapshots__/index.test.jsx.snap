// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trips should render 1`] = `
<Fragment>
  <div>
    <Map
      center={
        Array [
          -98.2518834,
          39.8112442,
        ]
      }
      mapClassName="this-is-the-map"
      points={
        Array [
          Array [
            -5.21714480341615,
            51.8780559,
          ],
          Array [
            -3.1936512,
            55.9499331,
          ],
          Array [
            2.1773041,
            41.3838014,
          ],
          Array [
            2.17439826359357,
            41.4034719,
          ],
        ]
      }
      zoom={1}
    />
  </div>
  <WithStyles(ForwardRef(Paper))
    elevation={3}
  >
    <PaperList
      actionButtonText="Add Trip"
      actionButtonUrl="/trips/new"
      thingsToList={
        Array [
          ThingToList {
            "date": "09/15/19 - 09/28/19",
            "id": 26,
            "title": "UK with Ron and Stacey",
            "url": "/trips/26",
          },
          ThingToList {
            "date": "09/01/19 - 09/15/19",
            "id": 25,
            "title": "Europe 2019",
            "url": "/trips/25",
          },
        ]
      }
      title="Trips"
    />
  </WithStyles(ForwardRef(Paper))>
</Fragment>
`;
